docs/schedule_scripting_language.md

```diff
@@ -0,0 +1,901 @@
+[miscellaneous commands](#miscellaneous-commands).
+
+### Generics and non-generics
+
+Schedule scripts use both [short and long commands](#short-and-long-commands)
+for most of the [conditional checks](#conditional-checks) and
+[unconditional branches](#unconditional-branches) commands, making it optimal
+space-wise, but terrible to worry about when actually writing a schedule script
+from the user's point of view.
+
+Due to this, this high level language allows for generic versions of those
+commands (a.k.a. suffix-less versions) that allow not having to worry about how
+many commands the body of a check has (and the byte length of them).
+
+The rest of this section will mostly refer to the generic versions of the
+commands. The non-generic versions of each command are available to be used too,
+but their use is not recommended.
+
+### Command arguments
+
+Some commands require arguments. Arguments are used by checking commands to
+check something specific of the state of the game (what's the current day? what's
+the current scene? etc.) and take a decision based on it (branch to another
+command).
+
+Arguments are enclosed in parentheses (`(` and `)`) and passed verbatim to the
+generated output, allowing a compiler for this language to not need to recognize
+any identifier used in arguments, allowing to use any custom identifier
+(specially useful for schedule result enums). Note this behaviour may
+change in a future version of the language.
+
+### Conditional checks
+
+Conditional checks commands are commands that take a decision based on
+the state of the game, allowing it to decide which set of subcommands to
+execute.
+
+All conditional checks require an argument and a body of subcommands. A
+conditional check can optionally be followed by an `else`.
+
+If the condition of a command is not satisfied and that command has
+an `else`, then control jumps to the body of that `else`. If
+there's no `else` and the condition isn't satisfied then control just
+falls through into the next command.
+
+#### `if_week_event_reg`
+
+Checks if the passed WeekEventReg flag is set, and execute the body of the
+command if it is set.
+
+##### `if_week_event_reg` arguments
+
+- Argument 0: A WeekEventReg flag. `WEEKEVENTREG_` macros are preferred.
+
+##### `if_week_event_reg` example
+
+```c
+if_week_event_reg (WEEKEVENTREG_61_02) {
+    return_s (31)
+}
+return_s (30)
+```
+
+##### `if_week_event_reg` non-generics
+
+`if_week_event_reg_s` and `if_week_event_reg_l`
+
+#### `if_time_range`
+
+Checks if the current time is between the passed time range.
+
+##### `if_time_range` arguments
+
+- Argument 0: Hour component of the start time
+- Argument 1: Minute component of the start time
+- Argument 2: Hour component of the end time
+- Argument 3: Minute component of the end time
+
+##### `if_time_range` example
+
+```c
+// Checks if the current time is between 18:00 ~ 6:00
+if_time_range (18, 0, 6, 0) {
+    return_time (18, 0, 6, 0, 2)
```

Comment:
Does supporting typical time syntax directly in the macro adds a lot of complexity?
like
```
if_time_range (18:00, 6:00) {
    return_time (18:00, 6:00, 2)
```
I'm sure there is some matching reason to keep the raw data format, but if possible I would like to consider readability here, even if it's just a redundant command.

Reply:
As mentioned in the other comment, the compiler doesn't process the arguments at all, they are passed verbatim to the generated macros, so adding something like this would indeed add a bit of complexity, but not that much I suppose. I would need to look into it